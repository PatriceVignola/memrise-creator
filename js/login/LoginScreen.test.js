/* @flow */

import React from 'react';
import { addNavigationHelpers, NavigationActions } from 'react-navigation';
import Renderer from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow';
import Login from './Login';

const shallowRenderer = new ShallowRenderer();

describe('Login', () => {
  const mockDispatchFunction = jest.fn();

  const navigation = addNavigationHelpers({
    state: {
      key: 'key',
      routeName: 'routeName',
      path: 'path',
    },
    dispatch: mockDispatchFunction,
  });

  test('renders a login page', () => {
    const shallow = shallowRenderer.render((
      <Login
        navigation={navigation}
      />
    ));
    expect(shallow).toMatchSnapshot();
  });

  test('navigates to GoogleLogin when clicking a course card', () => {
    const mockResetObject = NavigationActions.reset({
      index: 0,
      actions: [NavigationActions.navigate({ routeName: 'GoogleLogin' })],
    });

    const deep = Renderer.create((
      <Login
        navigation={navigation}
      />
    ));

    deep.root.findByProps({ title: 'Sign in with Google' }).props.onPress();

    expect(mockDispatchFunction).toHaveBeenCalledTimes(1);
    expect(mockDispatchFunction).toHaveBeenCalledWith(mockResetObject);
  });
});
