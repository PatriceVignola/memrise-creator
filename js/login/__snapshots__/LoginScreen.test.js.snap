// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login renders a login page 1`] = `
<Button
  onPress={[Function]}
  title="Sign in with Google"
/>
`;
